src/test/groovy/com/florianmski/suncalc/SunCalcSpec.groovy

```diff
@@ -93,5 +94,88 @@ class SunCalcSpec extends spock.lang.Specification {
+        d.setTimeZone(TimeZone.getTimeZone("UTC"))
+
+        when:
+        SunPosition actual = SunCalc.getSunPosition(d, 50.5, 30.5)
+
+        then:
+        // azimuth angle convention is 180 off, see https://github.com/mourner/suncalc/issues/6
+        near(actual.azimuth, -2.5003175907168385 + Math.PI)
+        near(actual.altitude, -0.7000406838781611)
+    }
+
+    /**
+     * port of original SunCalc-JS test
+     * https://github.com/mourner/suncalc/blob/master/test.js
+     */
+    def "getTimes returns sun phases for the given date and location: #description"() {
+
+        given:
+        double lat = 50.5
+        double lng = 30.5
+        DateFormat dtmFormat = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss'Z'")
+        DateFormat timeFormat = new SimpleDateFormat("HH:mm:ss")
+        TimeZone UTC = TimeZone.getTimeZone("UTC")
+        dtmFormat.setTimeZone(UTC)
+        timeFormat.setTimeZone(UTC)
+
+        and:
+        Date date = dtmFormat.parse(datetime)
+        Calendar d = Calendar.getInstance(UTC)
+        d.setTime(date)
+
+        when:
+        List<SunPhase> results = SunCalc.getPhases(d, lat, lng)
+
+        then:
+        List<SunPhase> actuals = results.findAll { it.name == sunPhaseName }
+        // sanity check
+        actuals.size == 1
+        SunPhase actual = actuals[0]
+
+        expect:
+        timeFormat.format(actual.startDate.time) == timeFormat.format(date)
+
+        where:
+        sunPhaseName                                | description     | datetime
+//        'TODO implement'      | 'nadir'         | '2013-03-04T22:10:57Z'
+        SunPhase.Name.TWILIGHT_ASTRONOMICAL_MORNING | 'nightEnd'      | '2013-03-05T02:46:17Z'
+        SunPhase.Name.TWILIGHT_NAUTICAL_MORNING     | 'nauticalDawn'  | '2013-03-05T03:24:31Z'
+        SunPhase.Name.TWILIGHT_CIVIL_MORNING        | 'dawn'          | '2013-03-05T04:02:17Z'
+        SunPhase.Name.SUNRISE                       | 'sunrise'       | '2013-03-05T04:34:56Z'
+        SunPhase.Name.GOLDEN_HOUR_MORNING           | 'sunriseEnd'    | '2013-03-05T04:38:19Z'
+        SunPhase.Name.DAYLIGHT                      | 'goldenHourEnd' | '2013-03-05T05:19:01Z'
+//        'TODO implement'      | 'solarNoon'     | '2013-03-05T10:10:57Z'
+        SunPhase.Name.GOLDEN_HOUR_EVENING           | 'goldenHour'    | '2013-03-05T15:02:52Z'
+        SunPhase.Name.SUNSET                        | 'sunsetStart'   | '2013-03-05T15:43:34Z'
+        SunPhase.Name.TWILIGHT_CIVIL_EVENING        | 'sunset'        | '2013-03-05T15:46:57Z'
+        SunPhase.Name.TWILIGHT_NAUTICAL_EVENING     | 'dusk'          | '2013-03-05T16:19:36Z'
+        SunPhase.Name.TWILIGHT_ASTRONOMICAL_EVENING | 'nauticalDusk'  | '2013-03-05T16:57:22Z'
+        SunPhase.Name.NIGHT_EVENING                 | 'night'         | '2013-03-05T17:35:36Z'
+
+
+    }
+
+    /** 1E-15 */
+    private static double DEFAULT_MARGIN = 0.000000000000001
+
+    private static boolean near(double val1, double val2) {
```

Comment:
i'm assuming that's the precision that the library is designed for? it was in the original JavaScript test ...